### PR TITLE
Correctly identify manifests which contain  more than 1 device

### PR DESCRIPTION
**Pull request by Flohack74**
This PR will add capabilities to correctly set up devices that are contained in a multi-device-manifest. You can basically do the following:

* Add a manifest containing common and device-specific repos for 2 devices and name it vendor_device1_device2.xml
* Start setup as usual with the device name
* It will check all xml files in the manifests folder to extract the vendor, and then make a substring compare on the remaining part

This will not affect that many devices, but it can help us keep manifests tidy.

### CODE REVIEW

File: setup [hunk 1]
```diff
@@ -81,7 +91,7 @@ else
     # Some device trees are lazy and contain more than one device. Check subdirectory with devicename, too
     # (e.g. Google Pixel)
     if [ -f $DEVICE_TREE/$DEVICE/setup-makefiles.sh ]; then
-        echo "I: Refreshing device vendor repository: device/$VENDOR/$DEVICE"
+        echo "I: Refreshing device vendor repository: device/$VENDOR/$DEVICE/$DEVICE"
```
Comment (z3ntu) [inline setup]: Having `$DEVICE` here twice looks suspicious

Comment (NotKit) [inline setup]: Is it expected that parent device and subdevice folder are named same? $DEVICE/$DEVICE looks a bit strange for me, would be nice to see some examples.

**Comment by Flohack74** [comment]
Yes true that log message made no sense ^^

**Comment by Flohack74** [comment]
Thats what it should look like:

grafik

**Pull request merged**